framework/test/mocha/unit/modules/chain/submodules/blocks/chain.js

```diff
@@ -591,19 +598,94 @@ describe('blocks/chain', () => {
 	});
 
 	describe('__private.applyConfirmedStep', () => {
-		/* eslint-disable mocha/no-pending-tests */
-		it('should return when block.transactions includes no transactions');
+		// Arrange
+		const filledBlock = {
+			id: 1,
+			height: 1,
+			transactions: [{ id: 21 }, { id: 22 }],
+		};
+
+		it('should return when block.transactions includes no transactions', async () => {
+			// Arrange
+			const emptyBlock = {
+				id: 1,
+				height: 1,
+				transactions: [],
+			};
+			// Act && Assert
+			return expect(__private.applyConfirmedStep(emptyBlock)).to.eventually.be
+				.undefined;
+		});
 
-		it('should call modules.processTransactions.undoTransactions');
+		it('should call modules.processTransactions.applyConfirmedStep', async () => {
+			// Arrange
+			const originalInertTransactions = global.exceptions.inertTransactions;
+			global.exceptions.inertTransactions = [filledBlock.transactions[0].id];
+			const dummyTx = {};
+
+			// Act
+			await __private.applyConfirmedStep(filledBlock, dummyTx);
+
+			// Cleanup
+			global.exceptions.inertTransactions = originalInertTransactions;
+
+			// Assert
+			return expect(
+				modules.processTransactions.applyTransactions
+			).to.have.been.calledWith([filledBlock.transactions[1]], dummyTx);
+		});
+
+		it('should throw error when errors exist in unappliedTransactionResponse', async () => {
+			// Arrange
+			const errors = [new Error('#Test Error')];
+			modules.processTransactions.applyTransactions.resolves({
+				...processTransactionMethodResponse,
+				transactionsResponses: [
+					{ status: TransactionStatus.OK },
+					{ status: TransactionStatus.FAIL, errors },
+				],
+			});
+
+			try {
+				// Act
+				await __private.applyConfirmedStep(filledBlock);
+			} catch (err) {
+				// Assert
+				expect(err).to.be.equal(errors);
+			}
+		});
+
+		it('should call stateStore.account.finalize', async () => {
+			// Act
+			await __private.applyConfirmedStep(filledBlock);
+
+			// Assert
+			return expect(
+				processTransactionMethodResponse.stateStore.account.finalize.calledOnce
```

Comment:
should be
`expect(processTransactionMethodResponse.stateStore.account.finalize).to.be.calledOnce`